xapian-core/weight/tfidfweight.cc

```diff
@@ -70,26 +74,48 @@ TfIdfWeight::name() const
 string
 TfIdfWeight::serialise() const
 {
-    return normalizations;
+    if (param_s > 0 || param_delta > 0) {
+	string result = serialise_double(param_s);
+	result += serialise_double(param_delta);
+	return result;
+    } else {
+	return normalizations;
+    }
 }
 
 TfIdfWeight *
-TfIdfWeight::unserialise(const string & s) const
+TfIdfWeight::unserialise(const string & str) const
 {
-    if (s.length() != 3)
-	throw Xapian::SerialisationError("Extra data in TfIdfWeight::unserialise()");
-    return new TfIdfWeight(s);
+    if (param_s > 0 || param_delta > 0) {
```

Comment:
Did you find [this document](https://xapian.org/docs/serialisation.html)? It explains what serialisation is used for, which might help.

For serialisation/unserialisation, you need a way of passing all the parameters you might need. I'd suggest putting the two doubles first, then using the rest as the normalisations string. (I don't believe it's a problem that this is different to previously, because I _think_ this is just a wire protocol, ie used at runtime. You could always check for a serialised string of length 3 and use that as normalisations, to support the old way if it's needed.)